packages/vm/src/imports.rs

```diff
@@ -266,12 +273,14 @@ pub fn do_next<S: Storage, Q: Querier>(
     let item = result?;
 
     // Prepare return values. Both key and value are Options and will be written if set.
-    let (key, value) = if let Some(result) = item {
-        let item = result?;
-        (Some(item.0), Some(item.1))
+    let ((key, value), used_gas) = if let Some(result) = item {
+        let (item, used_gas) = result?;
+        ((Some(item.0), Some(item.1)), used_gas)
     } else {
-        (Some(Vec::<u8>::new()), None) // Empty key will later be treated as _no more element_.
+        // TODO is gas charged for nonexistent keys?
```

Comment:
@webmaster128 do you know about this case? is gas charged for nonexistent keys when iterating?

Reply:
Looking at https://github.com/cosmos/cosmos-sdk/blob/v0.38.4/store/types/gas.go#L152-L163, I **suspect** there is a fee for the "next" call, which applies for the empty case as well.

Reply:
i see... what about just creating the iterator? I noticed that there's some funky logic around that in go-cosmwasm

Reply:
No idea. Ethan can help with that on Monday

Reply:
alright, cool

Reply:
I think I answered this in the other PR and you replied. But if still open, ping me. Otherwise, fee free to resolve